applicationinsights-react-js/src/ReactPlugin.ts

```diff
@@ -31,24 +37,27 @@ export default class ReactPlugin extends BaseTelemetryPlugin {
 
             _self.initialize = (config: IConfiguration & IConfig, core: IAppInsightsCore, extensions: IPlugin[], pluginChain?:ITelemetryPluginChain) => {
                 super.initialize(config, core, extensions, pluginChain);
-                _extensionConfig =
-                    config.extensionConfig && config.extensionConfig[_self.identifier]
-                        ? (config.extensionConfig[_self.identifier] as IReactExtensionConfig)
-                        : { history: null };
-        
-                arrForEach(extensions, ext => {
-                    const identifier = (ext as ITelemetryPlugin).identifier;
-                    if (identifier === 'ApplicationInsightsAnalytics') {
-                        _analyticsPlugin = (ext as any) as IAppInsights;
+
+                _self._addHook(onConfigChange(config, (details) => {
+                    let ctx = _self._getTelCtx();
+                    _extensionConfig = ctx.getExtCfg<IReactExtensionConfig>(this.identifier, defaultReactExtensionConfig);
+                    arrForEach(extensions, ext => {
+                        const identifier = (ext as ITelemetryPlugin).identifier;
+                        if (identifier === 'ApplicationInsightsAnalytics') {
+                            _analyticsPlugin = (ext as any) as IAppInsights;
+                        }
+                    });
+                    if (_extensionConfig.history) {
```

Comment:
I think we are going to need an additional check on whether this value has "changed".

The concern comes from the `trackPageView` function call below on line 58... We don't want to call `trackPageView` just because `onConfigChange` was called possible if the history object changed... So perhaps a flag indicating that we have already performed the trackPageView as well (or instead of the previous history value)

Reply:
I am thinking about that but do not know how to check whether _extensionConfig.history has been changed. If I store it somewhere, the copy item will also be updated when _extensionConfig.history is updated.

Reply:
Yeah, this is an interesting problem as it will depend.
- If it's a class then the value will be different (so we can just keep a local variable)
- If it's an object then (by default) it will become dynamic and therefore would be the same -- unless

There is now some settings for the defaults where we can specify that the property should not be converted see `cfgDfBlockPropValue()` which (should) then cause the 2nd option (if it's an object) to always be different.